### PR TITLE
issue #3: set version.snapshot to false if on a release tag without modifications

### PR DESCRIPTION
**Pull request by Vampire**

### CODE REVIEW

File: src/test/groovy/net/vivin/gradle/versioning/VersionUtilsTests.groovy [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import org.eclipse.jgit.lib.Repository
+import org.gradle.testfixtures.ProjectBuilder
+import org.testng.annotations.BeforeMethod
+import org.testng.annotations.Test
+
+import static org.testng.Assert.assertFalse
+
+class VersionUtilsTests extends TestNGRepositoryTestCase {
+
+    @Override
```
Comment (vivin) [inline src/test/groovy/net/vivin/gradle/versioning/VersionUtilsTests.groovy]: Is there a reason you overrode the setup method in `TestNGRepositoryTestCase`? I see that you create a temp directory and then clone into it. Was just curious as to why.

File: src/test/groovy/net/vivin/gradle/versioning/VersionUtilsTests.groovy [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    @Test
+    void testTaggedVersionIsRecognizedAsNonSnapshot() {
+        testRepository.commitAndTag("0.0.1")
+
+        SemanticBuildVersion version = (SemanticBuildVersion) project.getVersion()
+
+        version.versionUtils.determineVersion()
+        assertFalse(version.snapshot, "Tagged version is snapshot version")
```
Comment (vivin) [inline src/test/groovy/net/vivin/gradle/versioning/VersionUtilsTests.groovy]: Minor: I usually write assertions using "should" or "should not" so that it's clear what is expected. Maybe reword this to "Tagged version should not be snapshot version"?

**Comment by vivin** [comment]
👍 Thanks for the PR! Only a few minor comments. I will get this merged later today if possible.

File: src/test/groovy/net/vivin/gradle/versioning/VersionUtilsTests.groovy (hunk 1 above)
Comment (Vampire) [reply src/test/groovy/net/vivin/gradle/versioning/VersionUtilsTests.groovy]: Probably not, that was copy & paste from your TagTaskTest

File: src/test/groovy/net/vivin/gradle/versioning/VersionUtilsTests.groovy (hunk 2 above)
Comment (Vampire) [reply src/test/groovy/net/vivin/gradle/versioning/VersionUtilsTests.groovy]: As you wish
